teamcity: failed tests on master: testrace/TestMonotonicInserts/distsql=off, testrace/TestMonotonicInserts/distsql=on, testrace/TestMonotonicInserts, test/TestLogic, test/TestLogic/distsql, test/TestLogic/distsql/snapshot_issue2861, test/TestLogic, test/TestLogic/distsql, test/TestLogic/distsql/snapshot_issue2861, testrace/TestMonotonicInserts/distsql=off, testrace/TestMonotonicInserts/distsql=on, testrace/TestMonotonicInserts

The following tests appear to have failed:

[#485929](https://teamcity.cockroachdb.com/viewLog.html?buildId=485929):


```
--- FAIL: testrace/TestMonotonicInserts/distsql=off (0.000s)
Test ended in panic.

--- FAIL: testrace/TestMonotonicInserts/distsql=on (0.000s)
Race detected!

------- Stdout: -------
E180120 16:35:05.060025 6007 storage/queue.go:668  [replicate,n1,s1,r1/1:/{Min-System/}] range requires a replication change, but lacks a quorum of live replicas (0/1)
==================
Read at 0x00c4213ef230 by goroutine 520:
  reflect.Value.Bytes()
      /usr/local/go/src/reflect/value.go:259 +0x86
  encoding/gob.encUint8Array()
      /usr/local/go/src/encoding/gob/encode.go:241 +0x6a
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:328 +0x3ee
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:581 +0xed
  encoding/gob.(*Encoder).encodeArray()
      /usr/local/go/src/encoding/gob/encode.go:351 +0x313
  encoding/gob.encOpFor.func1()
      /usr/local/go/src/encoding/gob/encode.go:551 +0x170
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:328 +0x3ee
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:581 +0xed
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:328 +0x3ee
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:581 +0xed
  encoding/gob.(*Encoder).encodeArray()
      /usr/local/go/src/encoding/gob/encode.go:351 +0x313
  encoding/gob.encOpFor.func1()
      /usr/local/go/src/encoding/gob/encode.go:551 +0x170
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:328 +0x3ee
  encoding/gob.(*Encoder).encode()
      /usr/local/go/src/encoding/gob/encode.go:701 +0x24e
  encoding/gob.(*Encoder).EncodeValue()
      /usr/local/go/src/encoding/gob/encoder.go:250 +0x625
  encoding/gob.(*Encoder).Encode()
      /usr/local/go/src/encoding/gob/encoder.go:175 +0x72
  github.com/cockroachdb/cockroach/pkg/kv.GRPCTransportFactory.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport_race.go:83 +0x195
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:266 +0xf3

Previous write at 0x00c4213ef230 by goroutine 925:
  github.com/cockroachdb/cockroach/pkg/roachpb.MergeSpans()
      /go/src/github.com/cockroachdb/cockroach/pkg/roachpb/merge_spans.go:61 +0xa5a
  github.com/cockroachdb/cockroach/pkg/kv.(*TxnCoordSender).AugmentMeta()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn_coord_sender.go:272 +0x510
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).AugmentTxnCoordMeta()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:1093 +0x186
  github.com/cockroachdb/cockroach/pkg/sql.(*distSQLReceiver).Push()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:359 +0x20a
  github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go:159 +0xe3
  github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*noopProcessor).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/processors.go:494 +0xd6

Goroutine 520 (running) created at:
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:261 +0x14c
  github.com/cockroachdb/cockroach/pkg/kv.GRPCTransportFactory()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport_race.go:61 +0x300
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendToReplicas()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:1237 +0x165
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendRPC()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:382 +0x2ef
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendSingleRange()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:446 +0x1aa
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendPartialBatch()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:1027 +0x402
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).divideAndSendBatchToRanges()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:703 +0x1d3c
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:624 +0x539
  github.com/cockroachdb/cockroach/pkg/internal/client.lookupRangeFwdScan()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/range_lookup.go:296 +0x3ae
  github.com/cockroachdb/cockroach/pkg/internal/client.RangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/range_lookup.go:174 +0x130
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).RangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:286 +0xba
  github.com/cockroachdb/cockroach/pkg/kv.(*RangeDescriptorCache).performRangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:433 +0x421
  github.com/cockroachdb/cockroach/pkg/kv.(*RangeDescriptorCache).lookupRangeDescriptorInternal.func3()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:309 +0x187
  github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight/singleflight.go:118 +0x4c

Goroutine 925 (finished) created at:
  github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*Flow).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow.go:466 +0x675
  github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:209 +0xe78
  github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:468 +0x2a2
  github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execDistSQL()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:2030 +0x24d
  github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmt()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:2218 +0x699
  github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmtInOpenTxn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1866 +0x1082
  github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execSingleStatement()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1307 +0x9f2
  github.com/cockroachdb/cockroach/pkg/sql.runTxnAttempt()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1183 +0x4c3
  github.com/cockroachdb/cockroach/pkg/sql.runWithAutoRetry()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1008 +0x303
  github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execParsed()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:824 +0x32c
  github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execPrepared()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:693 +0x1af
  github.com/cockroachdb/cockroach/pkg/sql.(*Executor).ExecutePreparedStatement()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:670 +0x1ea
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*v3Conn).handleExecute()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/v3.go:875 +0x5e8
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*v3Conn).serve()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/v3.go:468 +0xadf
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).ServeConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/server.go:422 +0x11e0
  github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func17.1()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:1251 +0x17e
  github.com/cockroachdb/cockroach/pkg/util/netutil.(*Server).ServeWith.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/netutil/net.go:140 +0xdc
==================
--- FAIL: testrace/TestMonotonicInserts (0.000s)
Test ended in panic.

--- FAIL: test/TestLogic (53.080s)
test_log_scope.go:81: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic936907276
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: test/TestLogic/distsql (0.540s)
None
--- FAIL: test/TestLogic/distsql/snapshot_issue2861 (1.340s)
logic.go:2071: 
	testdata/logic_test/snapshot_issue2861:46: SELECT * FROM t
	expected:
	    1
	    2
	    
	but found (query options: "") :
	    2  
	    1  
	
logic.go:2102: testdata/logic_test/snapshot_issue2861:52: too many errors encountered, skipping the rest of the input
--- FAIL: test/TestLogic (53.080s)
test_log_scope.go:81: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic936907276
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: test/TestLogic/distsql (0.540s)
None
--- FAIL: test/TestLogic/distsql/snapshot_issue2861 (1.340s)
logic.go:2071: 
	testdata/logic_test/snapshot_issue2861:46: SELECT * FROM t
	expected:
	    1
	    2
	    
	but found (query options: "") :
	    2  
	    1  
	
logic.go:2102: testdata/logic_test/snapshot_issue2861:52: too many errors encountered, skipping the rest of the input
--- FAIL: testrace/TestMonotonicInserts/distsql=off (6.410s)
Test ended in panic.

--- FAIL: testrace/TestMonotonicInserts/distsql=on (6.560s)
Race detected!

------- Stdout: -------
E180120 16:35:05.060025 6007 storage/queue.go:668  [replicate,n1,s1,r1/1:/{Min-System/}] range requires a replication change, but lacks a quorum of live replicas (0/1)
==================
Read at 0x00c4213ef230 by goroutine 520:
  reflect.Value.Bytes()
      /usr/local/go/src/reflect/value.go:259 +0x86
  encoding/gob.encUint8Array()
      /usr/local/go/src/encoding/gob/encode.go:241 +0x6a
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:328 +0x3ee
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:581 +0xed
  encoding/gob.(*Encoder).encodeArray()
      /usr/local/go/src/encoding/gob/encode.go:351 +0x313
  encoding/gob.encOpFor.func1()
      /usr/local/go/src/encoding/gob/encode.go:551 +0x170
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:328 +0x3ee
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:581 +0xed
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:328 +0x3ee
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:581 +0xed
  encoding/gob.(*Encoder).encodeArray()
      /usr/local/go/src/encoding/gob/encode.go:351 +0x313
  encoding/gob.encOpFor.func1()
      /usr/local/go/src/encoding/gob/encode.go:551 +0x170
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:328 +0x3ee
  encoding/gob.(*Encoder).encode()
      /usr/local/go/src/encoding/gob/encode.go:701 +0x24e
  encoding/gob.(*Encoder).EncodeValue()
      /usr/local/go/src/encoding/gob/encoder.go:250 +0x625
  encoding/gob.(*Encoder).Encode()
      /usr/local/go/src/encoding/gob/encoder.go:175 +0x72
  github.com/cockroachdb/cockroach/pkg/kv.GRPCTransportFactory.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport_race.go:83 +0x195
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:266 +0xf3

Previous write at 0x00c4213ef230 by goroutine 925:
  github.com/cockroachdb/cockroach/pkg/roachpb.MergeSpans()
      /go/src/github.com/cockroachdb/cockroach/pkg/roachpb/merge_spans.go:61 +0xa5a
  github.com/cockroachdb/cockroach/pkg/kv.(*TxnCoordSender).AugmentMeta()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn_coord_sender.go:272 +0x510
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).AugmentTxnCoordMeta()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:1093 +0x186
  github.com/cockroachdb/cockroach/pkg/sql.(*distSQLReceiver).Push()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:359 +0x20a
  github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go:159 +0xe3
  github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*noopProcessor).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/processors.go:494 +0xd6

Goroutine 520 (running) created at:
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:261 +0x14c
  github.com/cockroachdb/cockroach/pkg/kv.GRPCTransportFactory()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport_race.go:61 +0x300
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendToReplicas()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:1237 +0x165
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendRPC()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:382 +0x2ef
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendSingleRange()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:446 +0x1aa
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendPartialBatch()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:1027 +0x402
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).divideAndSendBatchToRanges()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:703 +0x1d3c
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:624 +0x539
  github.com/cockroachdb/cockroach/pkg/internal/client.lookupRangeFwdScan()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/range_lookup.go:296 +0x3ae
  github.com/cockroachdb/cockroach/pkg/internal/client.RangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/range_lookup.go:174 +0x130
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).RangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:286 +0xba
  github.com/cockroachdb/cockroach/pkg/kv.(*RangeDescriptorCache).performRangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:433 +0x421
  github.com/cockroachdb/cockroach/pkg/kv.(*RangeDescriptorCache).lookupRangeDescriptorInternal.func3()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:309 +0x187
  github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight/singleflight.go:118 +0x4c

Goroutine 925 (finished) created at:
  github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*Flow).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow.go:466 +0x675
  github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:209 +0xe78
  github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:468 +0x2a2
  github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execDistSQL()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:2030 +0x24d
  github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmt()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:2218 +0x699
  github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmtInOpenTxn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1866 +0x1082
  github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execSingleStatement()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1307 +0x9f2
  github.com/cockroachdb/cockroach/pkg/sql.runTxnAttempt()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1183 +0x4c3
  github.com/cockroachdb/cockroach/pkg/sql.runWithAutoRetry()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1008 +0x303
  github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execParsed()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:824 +0x32c
  github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execPrepared()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:693 +0x1af
  github.com/cockroachdb/cockroach/pkg/sql.(*Executor).ExecutePreparedStatement()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:670 +0x1ea
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*v3Conn).handleExecute()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/v3.go:875 +0x5e8
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*v3Conn).serve()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/v3.go:468 +0xadf
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).ServeConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/server.go:422 +0x11e0
  github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func17.1()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:1251 +0x17e
  github.com/cockroachdb/cockroach/pkg/util/netutil.(*Server).ServeWith.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/netutil/net.go:140 +0xdc
==================
------- Stdout: -------
E180120 16:34:00.699202 18451 storage/consistency_queue.go:111  [replica consistency checker,n3,s3,r11/3:/Table/1{4-5}] r11 was not found
--- FAIL: testrace/TestMonotonicInserts (13.010s)
Test ended in panic.


```

Please assign, take a look and update the issue accordingly.

--- a/pkg/kv/txn_coord_sender.go
+++ b/pkg/kv/txn_coord_sender.go
@@ -269,7 +269,10 @@ func (tc *TxnCoordSender) AugmentMeta(meta roachpb.TxnCoordMeta) {
 		return
 	}
 	tc.mu.meta.Txn.Update(&meta.Txn)
-	tc.mu.meta.Intents, _ = roachpb.MergeSpans(append(tc.mu.meta.Intents, meta.Intents...))
+	// Do not modify existing span slices when copying.
+	tc.mu.meta.Intents, _ = roachpb.MergeSpans(
+		append(append([]roachpb.Span(nil), tc.mu.meta.Intents...), meta.Intents...),
+	)
 	tc.mu.meta.CommandCount += meta.CommandCount
 
 	// Recompute the size of the intents.

